### PR TITLE
Cleanup CIs

### PR DESCRIPTION
**Pull request by j8r**
There are 36 CI jobs for each commit landing in master (!). Obviously, it has a visibility impact for the users, and unnecessary use resources.

After looking at them, some can be removed:
- Travis CI
- check_format on Circle CI

Be free for core members to comment on this PR.

Note: I don't see in the Circle CI config how to disable the jobs running on a push/PR, maybe it is available on the Web console.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 # Crystal
 
-[![Travis CI Build Status](https://travis-ci.org/crystal-lang/crystal.svg)](https://travis-ci.org/crystal-lang/crystal)
+![Linux CI](https://github.com/crystal-lang/crystal/workflows/Linux%20CI/badge.svg)
+![macOS CI](https://github.com/crystal-lang/crystal/workflows/macOS%20CI/badge.svg)
+![AArch64 CI](https://github.com/crystal-lang/crystal/workflows/AArch64%20CI/badge.svg)
+![Windows CI](https://github.com/crystal-lang/crystal/workflows/Windows%20CI/badge.svg)
```
Comment (oprypin) [inline README.md:6]: These would ideally be wrapped into links, such as https://github.com/crystal-lang/crystal/actions?query=workflow%3A%22Windows+CI%22+event%3Apush+branch%3Amaster

Comment (j8r) [reply README.md:6]: One little issue is all fork runs are listed.

**Comment by straight-shoota** [comment]
We can probably remove the entire `test_all_platforms` workflow on circleci and only keep the release related workflows.

**Comment by j8r** [comment]
If I understand correctly the logic, we can remove all `test_*` from Circle CI and keep the `dist_*`? At least, some like Linux and preview_mt.

One thing remaining is to change the env var `TRAVIS_OS_NAME` naming.

**Review by bcardiff — CHANGES_REQUESTED** [review]
This is PR as it is is dropping the publishing of docs on each master commit. We can add the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` secrets to github but the appropriate job would still be needed.

**Comment by j8r** [comment]
Where is the script that does it? I don't see it.
Also, I need the S3 bucket in question.

**Comment by straight-shoota** [comment]
Oh, I didn't even realizes that dev docs are actually updated after each commit to master. I had in mind that this happens with the nightly release.

**Comment by straight-shoota** [comment]
Docs are built in test_linux job (`bin/ci` runs `make docs`) and then `sync_docs_s3` is responsible for pushing it to s3.

**Comment by j8r** [comment]
Thanks @straight-shoota , Circle CI configuration is quite powerful!

What do we do, do we move `test_all_platforms` to GitHub Actions at the end? It depends if the goal is also to move progressively from Circle CI to Actions.

**Review by straight-shoota — DISMISSED** [review]

**Comment by straight-shoota** [comment]
Which parts of `test_all_platforms` is currently not in Github Actions?

**Comment by j8r** [comment]
`sync_docs_to_s3` is not in GitHub Actions, that's it.

**Comment by straight-shoota** [comment]
Okay, but that's in this PR. So there's nothing left for "move test_all_platforms to GitHub Actions at the end".

**Review by straight-shoota — APPROVED** [review]

**Review by bcardiff — APPROVED** [review]
I iterated a bit in the configuration. I restored the commented test_darwin because they are disabled temporarily. I also moved the `if` to job level.

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**